packages/components/src/unit-control/index.tsx

```diff
@@ -165,7 +166,10 @@ function UnforwardedUnitControl(
 		}
 	};
 
-	const handleOnBlur: FocusEventHandler< HTMLInputElement > = mayUpdateUnit;
+	const handleOnBlur: FocusEventHandler< HTMLInputElement > = ( event ) => {
+		mayUpdateUnit( event );
```

Comment:
I thought about this, and I'm not sure it would make any difference:

- changing the `unit` from the select dropdown already fires the `onUnitChange` and `onChange` callbacks
- typing a unit in the `NumberControl` input is only allowed when the `isPressEnterToChange` prop is set to `true` — which is already the logic implemented in `mayUpdateUnit`

Maybe I'm missing something?

Reply:
That's right, `mayUpdateUnit` is only to check if a unit can be parsed from the number input and intentionally limited to `isPressEnterToChange` being `true`.

Reply:
Ok, thanks. That makes sense to me 👍